tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/EquinoxResolver.java

```diff
@@ -194,15 +215,25 @@ public ModuleContainer newResolvedState(ReactorProject project, MavenSession mav
             return container;
         } finally {
             executorService.shutdownNow();
+            if (!config.keepUses) {
+                logger.info(FORCE_KEEP_USES);
+                config.forceKeepUses();
```

Comment:
If we really want to go with a config object, it should be final and unmodified and we create a new object with adjusted settings here...

Reply:
you mean a record? Are we fine moving to newer Java allowing records? I don't remember if we agreed on something for Tycho?

Reply:
Just use a traditional final class with package protected final fields, getters or whats you preferred pattern, we can convert ti to records later on :-)